kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -0,0 +1,729 @@
+}
+
+template <>
+inline void EmbeddedNodalVariableFromSkinTypeHelperClass<double>::AddUnknownVariable(ModelPart &rModelPart)
+{
+    rModelPart.AddNodalSolutionStepVariable(NODAL_MAUX);
+}
+
+template <>
+inline void EmbeddedNodalVariableFromSkinTypeHelperClass<array_1d<double,3>>::AddUnknownVariable(ModelPart &rModelPart)
+{
+    rModelPart.AddNodalSolutionStepVariable(NODAL_VAUX);
+}
+
+template <>
+inline void EmbeddedNodalVariableFromSkinTypeHelperClass<double>::AddUnknownVariableDofs(ModelPart &rModelPart)
+{
+    VariableUtils().AddDof(NODAL_MAUX, rModelPart);
+}
+
+template <>
+inline void EmbeddedNodalVariableFromSkinTypeHelperClass<array_1d<double, 3>>::AddUnknownVariableDofs(ModelPart &rModelPart)
+{
+    VariableUtils().AddDof(NODAL_VAUX_X, rModelPart);
+    VariableUtils().AddDof(NODAL_VAUX_Y, rModelPart);
+    VariableUtils().AddDof(NODAL_VAUX_Z, rModelPart);
+}
+
+template <class TVarType, class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class KRATOS_API(KRATOS_CORE) CalculateEmbeddedNodalVariableFromSkinProcess : public Process
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef typename Element::NodeType::Pointer NodePointerType;
+    typedef typename Scheme<TSparseSpace,TDenseSpace>::Pointer SchemePointerType;
+    typedef typename BuilderAndSolver<TSparseSpace,TDenseSpace,TLinearSolver>::Pointer BuilderSolverPointerType;
+    typedef typename SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::UniquePointer SolvingStrategyPointerType;
+    typedef typename FindIntersectedGeometricalObjectsProcess::Pointer FindIntersectedGeometricalObjectsProcessPointerType;
+
+    struct Hash
+    {
+        std::size_t operator()(const std::pair<unsigned int,bool>& k) const
+        {
+            std::size_t h1 = std::hash<unsigned int>()(std::get<0>(k));
+            std::size_t h2 = std::hash<bool>()(std::get<1>(k));
+            return h1 ^ (h2 << 1);
+        }
+    };
+
+    struct KeyEqual
+    {
+        bool operator()(const std::pair<std::size_t, std::size_t>& lhs, const std::pair<std::size_t, std::size_t>& rhs) const
+        {
+            return ((std::get<0>(lhs) == std::get<0>(rhs)) && (std::get<1>(lhs) == std::get<1>(rhs)));
+        }
+    };
+
+    typedef std::unordered_map<std::pair<std::size_t, std::size_t>, std::size_t, Hash, KeyEqual> EdgesMapType;
+
+    ///@}
+    ///@name Pointer Definitions
+
+    /// Pointer definition of CalculateEmbeddedNodalVariableFromSkinProcess
+    KRATOS_CLASS_POINTER_DEFINITION(CalculateEmbeddedNodalVariableFromSkinProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    CalculateEmbeddedNodalVariableFromSkinProcess(
+        ModelPart &rBaseModelPart,
+        ModelPart &rSkinModelPart,
+        const Variable<TVarType> &rSkinVariable,
+        const Variable<TVarType> &rEmbeddedNodalVariable,
+        const std::string LevelSetType = "continuous",
+        const std::string AuxPartName = "IntersectedElementsModelPart")
```

Comment:
```suggestion
        const std::string& AuxPartName = "IntersectedElementsModelPart")
```

Reply:
(same comment)

Reply:
basic types such as int and double yes, but I think string is treated separately
at least codacy says so

Reply:
Honestly, I'm not really sure about this. So just in case I'd leave it as it is.